### DISCUSSION TITLE
TypeError: app.modal.confirm is not a function

### DISCUSSION THREAD

**Discussion by adithyaharun** [ant-design/ant-design#47455] [Q&A]
Tried to use `useApp` hook on React project from `create-react-app`. Got an exception like above. Might be throwing the same exception if using `message` and `notification` since all methods are not exists. Please take a look at the attachments for details.

React: `^18.2.0`
Antd: `^5.14.1`

![image](https://github.com/ant-design/ant-design/assets/7106825/751c1e3d-ba03-4cf2-aa01-94c3169cdfc3)
![image](https://github.com/ant-design/ant-design/assets/7106825/577a8551-dd4b-4e60-8b36-916747847f13)

**Comment by Liuhuifeng** [comment]
same issue

**Comment by gguridi** [comment]
The useApp seems that's not working anymore. You can do the same with this, even though it is not consistent:

```
import { Modal, notification } from 'antd';

const Component = () => {
    Modal.info({content: "foo"})
    notification.info({content: "bar"})
}
```
